process_report/process_report.py

```diff
@@ -1,9 +1,97 @@
 import argparse
 import os
+import sys
 
+import json
 import pandas
 
 
+### Invoice field names
+INVOICE_DATE_FIELD = 'Invoice Month'
+PROJECT_FIELD = 'Project - Allocation'
+PROJECT_ID_FIELD = 'Project - Allocation ID'
+PI_FIELD = 'Manager (PI)'
+INVOICE_EMAIL_FIELD = 'Invoice Email'
+INVOICE_ADDRESS_FIELD = 'Invoice Address'
+INSTITUTION_FIELD = 'Institution'
+INSTITUTION_ID_FIELD = 'Institution - Specific Code'
+SU_HOURS_FIELD = 'SU Hours (GBhr or SUhr)'
+SU_TYPE_FIELD = 'SU Type'
+COST_FIELD = 'Cost'
+CREDIT_FIELD = 'Credit'
+CREDIT_CODE_FIELD = 'Credit Code'
+BALANCE_FIELD = 'Balance'
+###
+
+
+def apply_credits_0001(dataframe):
+    credit_code = "0001"
+    pass
+
+
+def apply_credits_0002(dataframe):
+    """Applies the New PI Credit. This credit function expects the 
+    env var `C0002_OLD_PI` to be set, pointing to a txt file containing old PIs"""
+    credit_code = "0002"
+    credit_amount = 1000
+
+    old_pi_list = set()
+    if "C0002_OLD_PI" not in os.environ:
+        print("Applying credit 0002 failed. C0002_OLD_PI env var is not set")
+        sys.exit(1)
+
+    try:
+        with open(os.getenv("C0002_OLD_PI")) as f:
+            for pi in f: old_pi_list.add(pi.strip())
+
+    except FileNotFoundError:
+        print("Applying credit 0002 failed. Old PI file does not exist")
+        sys.exit(1)
+
+    print("Old pi list: ", old_pi_list)
+    pi_list = dataframe[PI_FIELD].unique()
+
+    for pi in pi_list:
+        if pandas.isna(pi): 
+            continue   # NaN check
+        if pi in old_pi_list: 
+            continue  # Is the PI an old PI?
+
+        pi_projects = dataframe[dataframe[PI_FIELD] == pi]
+        rem_credit = credit_amount
+        for i, row in pi_projects.iterrows():
+            project_cost = row[COST_FIELD]
+            if project_cost >= rem_credit:
+                dataframe.at[i, CREDIT_FIELD] =  rem_credit
```

Comment:
@joachimweyl @QuanMPhm I am slightly confused by this. Is our desired outcome that for a PI that has multiple projects and a credit, the credit field in the CSV decrease row by row as the credits are exhausted or do we want the credit field in all rows to be the final computation?

Reply:
this is based on a Google sheet that the easiest way to do it was to decrease row by row, that being said it does not matter how it is done as long as we reduce by $100 or total whichever is smaller.

Reply:
@joachimweyl $100?

Reply:
sorry $1000, missed a zero.

Reply:
Then I would strongly recommend not decreasing the credit amount line by line but making sure all cells for a PI are correct.

Reply:
The code block confused me as it seem to save applied credit to the cell, rather than remaining credit. 

I suggested Quan to rewrite it to remove the if-else logic and use something like 
```python
>>> applied_credit = min(project_cost, remaining_credit)
>>> remaining_credit -= applied_credit
```

Reply:
The code block has been rewritten.